Jenkinsfile

```
@@ -0,0 +1,106 @@
+@Library('jenkins.shared.library') _
+
+pipeline {
+  agent {
+    label 'ubuntu_docker_label'
+  }
+  tools {
+    go "Go 1.13"
+  }
+  options {
+    checkoutToSubdirectory('src/github.com/infobloxopen/ingress-nginx')
+  }
+  environment {
+    DIRECTORY = "src/github.com/infobloxopen/ingress-nginx"
+    GIT_VERSION = sh(script: "cd ${DIRECTORY} && git describe --always --long --tags",
+                       returnStdout: true).trim()
+    TAG = "${env.GIT_VERSION}-j${env.BUILD_NUMBER}"
+    REGISTRY = 'infoblox'
+    IMGNAME = 'nginx-fips'
+    PLATFORMS = 'amd64'
+    ARCH = 'amd64'
+    MULTI_ARCH_IMG="${env.REGISTRY}/${env.IMGNAME}-${env.ARCH}"
+    MULTI_ARCH_IMAGE = "${env.REGISTRY}/${env.IMGNAME}"
+    BASEIMAGE = "${REGISTRY}/${env.IMGNAME}:${TAG}"
+    DOCKER_CLI_EXPERIMENTAL = 'enabled'
+    GOPATH = "$WORKSPACE"
+  }
+  stages {
+    stage("Setup docker-ce") {
+        steps {
+          sh "curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -"
+          sh 'sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"'
+          sh "sudo apt-get update"
+          sh 'sudo apt-get -y -o Dpkg::Options::="--force-confnew" install docker-ce'
+        }
+    }
+    stage("Setup") {
+      steps {
+        prepareBuild()
+      }
+    }
+    stage("Build NGINX Image") {
+      steps {
+        dir("$DIRECTORY/images/nginx") {
+          sh "make container"
+          sh "make fips-test"
+        }
+      }
+    }
+    stage("Push NGINX Image") {
+      when {
+        anyOf { branch 'nginx-0.27.1-fips'; buildingTag() }
+      }
+      steps {
+        withDockerRegistry([credentialsId: "${env.JENKINS_DOCKER_CRED_ID}", url: ""]) {
+          dir("$DIRECTORY/images/nginx") {
+            sh "docker push ${env.REGISTRY}/${env.IMGNAME}:${env.TAG}"
+          }
+        }
+      }
+    }
+    stage("Ingress Unit Tests") {
+      steps {
+        dir("$DIRECTORY") {
+          sh "make test"
+        }
+      }
+    }
+    stage("Build Ingress Image") {
+      steps {
+        withEnv(["TAG=ingress-${env.TAG}"]) {
+          dir("$DIRECTORY") {
+            sh "make build && make build-plugin"
+            sh "make container"
+          }
+        }
+      }
+    }
+    stage("Push Ingress Image") {
+      when {
+        anyOf { branch 'nginx-0.27.1-fips'; buildingTag() }
```

This might be better as a pattern... Do you expect to use a new branch like `nginx-0.42-fips` when upgrading to ingress-controller 0.42? If you want to avoid changing the pipeline later, you could do it like this:
```suggestion
        anyOf { branch 'nginx-0.*-fips'; buildingTag() }
```